tensorflow_model_optimization/python/core/sparsity_tf2/sparse_utils_test.py

```diff
@@ -0,0 +1,164 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for the utilities in sparse training pipelines."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl.testing import parameterized
+import numpy as np
+import tensorflow as tf
+
+from tensorflow.python.keras import keras_parameterized
+from tensorflow_model_optimization.python.core.sparsity_tf2 import sparse_utils
+
+dtypes = tf.dtypes
+test = tf.test
+
+
+def sample_noise(shape=(), mu=0, sigma=1.):
+  sample = tf.random.normal(shape, mean=mu,  stddev=sigma, dtype=tf.float64)
+  return sample
+
+def calculate_stdev(p):
+  return np.sqrt(p * (1 - p))
+
+class SparseUtilsTest(test.TestCase, parameterized.TestCase):
+
+  def setUp(self):
+    super(SparseUtilsTest, self).setUp()
+    self.ratios = [0.0, 0.1, 0.25, 0.5, 0.75, 0.9, 1.0]
+    self.shapes = [(), [1], [5], [2, 2], [4, 4], [8, 4], [3, 17], [3, 6, 9]]
+    self.seeds = range(20)
+
+  def matrix_init(self, _shape):
+    return tf.ones(_shape)
+  
+  def matrix_init_with_type(self, _shape, _type):
+    tf.ones(_shape, dtype=_type)
+
+  def get_bernouilli(self, p):
```

Comment:
Inline this helper. It doesn't buy much syntactic sugar and makes the tests harder to interpret. E.g. if you added extra logic to this helper it would no longer match the `Bernoulli` API and your test would be testing the helper, not your `Bernoulli` method.